### PR TITLE
feat(unlock-app) - handle stripe account that are not ready to be charged

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description
Our UI is not handling the case of a Stripe account with pending verification.

For this specific case, we now show a warning and a button to resume the Stripe setup.
Otherwise, we can also disconnect the Stripe connection from the Lock and choose, for example, another one from the list of Stripe accounts. 

**Stripe account not ready to be charged** 

Screenshot 2023-05-26 at 17 57 14

**Disconnecting account** 

Screenshot 2023-05-26 at 17 58 24

**Stripe account disconnected** 

Screenshot 2023-05-26 at 18 22 40

# Issues

Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [x] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx [hunk 1]
```diff
@@ -258,14 +241,79 @@ const ConnectStripe = ({
...
+}: Pick<ConnectStripeProps, 'isManager' | 'disabled'> &
+  ConnectStripe & {
+    connectedStripeAccount?: string
+  }) => {
+  return (
+    <span className="grid gap-2 text-sm">
+      <span className="font-semibold text-red-500">
+        Stripe account is connected, but not ready to process charges. Make sure
+        that all information are validated and you email is verified from
+        Stripe.
```
Comment (kalidiagne) [inline unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx:259]: @julien51  can be improved?

Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx:259]:
```suggestion
        Your Stripe account is connected but not ready to process charges yet. Make sure
        that all the details you entered on Stripe are valid and your email has been verified from
        Stripe.
```

**Review by julien51 — APPROVED** [review]
This is a good approach!

File: unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx [hunk 2]
```diff
@@ -294,54 +342,75 @@ export const CreditCardForm = ({
...
+        onSuccess: (connect: any) => {
+          if (connect?.url) {
+            window.location.assign(connect.url)
+          } else {
+            ToastHelper.success('Stripe connection succeeded!')
+          }
+        },
+        onError: () => {
+          ToastHelper.error('Stripe connection failed')
+        },
```
Comment (searchableguy) [inline unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx:373]: we should stop relying on inline onError and onSuccess. They are going to be removed in react-query-v4

**Review by searchableguy — APPROVED** [review]

**Pull request merged**